contracts/staking/test/unit_tests/lib_proxy.ts

```diff
@@ -0,0 +1,324 @@
+import { blockchainTests, constants, expect, hexRandom } from '@0x/contracts-test-utils';
+import { StakingRevertErrors } from '@0x/order-utils';
+
+import { artifacts, TestLibProxyContract, TestLibProxyReceiverContract } from '../../src';
+
+blockchainTests.resets('LibProxy', env => {
```

Comment:
We might want to have folders in the `unit_tests` folder to match the directory structure of `staking/contracts/src/` so that the `unit_tests` folder doesn't get to cluttered, but I don't feel to strongly one way or the other

Reply:
I thought about that, but I don't think it's necessary for now. We can revisit this later.